### DISCUSSION TITLE
Reduce import cost of TSDX package

### DISCUSSION THREAD

**Discussion by glenngijsberts** [jaredpalmer/tsdx#991] [Q&A]
Hey 👋 

I'm looking into improving the bundle size & performance of my app and while I was investigating that I noticed that the import cost of components from my components library (written in TSDX) is a lot compared to other libraries I've been using. I'm kinda new to this kind of performance improvements so I was wondering if anybody has some advice for me on what to do or If this is a normal import cost.

Examples

<img width="1248" alt="Screenshot 2021-03-14 at 22 09 50" src="https://user-images.githubusercontent.com/14276144/111084379-6415fc00-8512-11eb-8ded-2d6896077d00.png">
<img width="793" alt="Screenshot 2021-03-14 at 22 12 22" src="https://user-images.githubusercontent.com/14276144/111084375-5c565780-8512-11eb-99c1-98b139bc9389.png">
<img width="701" alt="Screenshot 2021-03-14 at 22 14 27" src="https://user-images.githubusercontent.com/14276144/111084435-a6d7d400-8512-11eb-8c2e-d712bcf63a03.png">

Looking at the pictures above it seems that if I import a single thing it's still costing me a lot and it seems that there is a lot more imported besides the single function/component. Is there a way I can reduce this import cost? Can I split those components in separate build files somehow? Any other tips on improving this? I'm using the standard setup (without a tsdx.config.js file).

Thanks in advance!

**Comment by gauravverma029** [comment]
Did you resolved this issue ?

**Reply by bryanprimus** [reply]
any update on this?

**Comment by robertdb** [comment]
Did you resolved this issue ?
